pkg/envoy/route/route_config.go

```diff
@@ -186,6 +187,9 @@ func buildRoute(pathMatchTypeType trafficpolicy.PathMatchType, path string, meth
 				ClusterSpecifier: &xds_route.RouteAction_WeightedClusters{
 					WeightedClusters: buildWeightedCluster(weightedClusters, totalWeight, direction),
 				},
+				HostRewriteSpecifier: &xds_route.RouteAction_HostRewriteLiteral{
+					HostRewriteLiteral: hostHeader,
+				},
```

Comment:
I missed this change. We should not be rewriting the host header in any way here. This needs to be removed.